flow/core/kernel/power_demand.py

```diff
@@ -0,0 +1,64 @@
+import math
+import random
+import statistics
+import numpy as np
+from scipy.interpolate import interp1d
+from collections import namedtuple
+from flow.energy.base_energy import BaseEnergyModel
+from abc import ABCMeta, abstractmethod
+
+class PowerDemandModel(BaseEnergyModel, metaclass=ABCMeta):
+    """Calculate power consumption of a vehicle.
+    Assumes vehicle is an average sized vehicle.
+    The power calculated here is the lower bound of the 
+    actual power consumed by the vehicle
+    """
+
+    def __init__(self,kernel):
+        self.k = kernel
+        self.k.env.vehicle = vehicle

```

Comment:
At the top of these child init's, you can inherit the parent class init by doing:
`super(BaseEnergyModel, self).__init__(<fill in call signature here>)`